javaparser-core/src/main/java/com/github/javaparser/ast/body/BodyDeclaration.java

```diff
@@ -124,4 +125,191 @@ public boolean replace(Node node, Node replacementNode) {
         }
         return super.replace(node, replacementNode);
     }
+
+    @Generated("com.github.javaparser.generator.core.node.TypeCastingGenerator")
+    public boolean isAnnotationDeclaration() {
+        return false;
+    }
+
+    @Generated("com.github.javaparser.generator.core.node.TypeCastingGenerator")
+    public AnnotationDeclaration asAnnotationDeclaration() {
+        return (AnnotationDeclaration) this;
```

Comment:
should this throw an exception?

Reply:
I think ClassCastException would be acceptable, however I think the code could be slightly surprising: a causal reader could wonder why we try to cast if we are sure the cast is going to fail. We could make it explicit that this method will always fail

Reply:
Would you accept...
- base method only throws an `IllegalStateException` (because we *do* support this operation, but not in this specific state)
- specific node returns `this`.

So more or less what you have but another exception?

Reply:
Latest commit does that ^^^^

Reply:
Yes, that would be great